log/logger.go

```diff
@@ -0,0 +1,97 @@
+// Package log provides sane default loggers using slog.
+package log
+
+import (
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+const DefaultRefreshInterval = time.Second * 5
+
+// NewDefaultEnvLogger creates new slog.Logger using sane default configuration and sets it as a default logger.
+// Environment variables can be used to configure loggers format and level. Changing log level at runtime is also supported.
+//
+// Name:			Value:
+// LOG_LEVEL		DEBUG|INFO|WARN|ERROR
+// LOG_FORMAT		JSON|TEXT
+//
+// Note: LOG_FORMAT can't be changed at runtime.
+func NewDefaultLogger() *slog.Logger {
+	lvl := &slog.LevelVar{}
+	lvl.Set(ParseLogLevelFromEnv())
+	go RefreshLogLevel(lvl, time.NewTicker(DefaultRefreshInterval))
+
+	handlerFn := ParseFormatEnv()
+	opts := &slog.HandlerOptions{
+		AddSource: true,
+		Level:     lvl,
+	}
+
+	logger := slog.New(handlerFn(os.Stdout, opts))
+	slog.SetDefault(logger)
+
+	return logger
+}
```

Comment:
Would we like to also have a `NewLogger` utility handler with which we could set the desired log level as a function argument, or leave the implementation of a new logger with custom log level to the user? Right now it is fetched from the environment which can be later set with `RefreshLogLevel`. Should we allow users also set log level programmatically on logger initialization?

Reply:
I feel like there's slog already gives the nice building blocks for that so there should be no need for that. Also I don't see usecase for setting log level manually or atleast haven't spotted it anywhere in our code base yet.